src/secrets/CommandEnv.ts

```diff
@@ -26,297 +27,324 @@ class CommandEnv extends CommandPolykey {
+      const preservedSecrets = new Set<string>();
+      for (const [vaultName, secretPath] of options.preserveNewline) {
+        // The vault name is guaranteed to have a value.
+        // If a secret path is undefined, then the newline preservation was
+        // targeting the secrets of the entire vault. Otherwise, the target
+        // was a single secret.
+        if (secretPath == null) preservedSecrets.add(vaultName);
+        else preservedSecrets.add(`${vaultName}:${secretPath}`);
+      }
+      // There are a few stages here
+      // 1. parse the desired secrets
+      // 2. obtain the desired secrets
+      // 3. switching behaviour here based on parameters
+      //   a. exec the command with the provided env variables from the secrets
+      //   b. output the env variables in the desired format
+
+      // Emulate commander's parsing to allow for parsing --
+      const fullArgs = process.argv;
+      const fullSeparatorIndex = fullArgs.indexOf('--');
+      const envVariables: Array<ParsedSecretPathValue> = [];
+      let cmd: string | undefined;
+      let argv: Array<string> = [];
+      if (fullSeparatorIndex === -1) {
+        // If the separator exists, then treat all args as environment paths
+        for (const arg of args) {
+          envVariables.push(binParsers.parseSecretPathEnv(arg));
         }
-        // There are a few stages here
-        // 1. parse the desired secrets
-        // 2. obtain the desired secrets
-        // 3. switching behaviour here based on parameters
-        //   a. exec the command with the provided env variables from the secrets
-        //   b. output the env variables in the desired format
-        const [envVariables, [cmd, ...argv]] = args;
-        const clientOptions = await binProcessors.processClientOptions(
-          options.nodePath,
-          options.nodeId,
-          options.clientHost,
-          options.clientPort,
-          this.fs,
-          this.logger.getChild(binProcessors.processClientOptions.name),
-        );
-        const meta = await binProcessors.processAuthentication(
-          options.passwordFile,
-          this.fs,
+      } else {
+        // Otherwise, separate command from environment paths
+        const afterSeparatorCount = fullArgs.length - fullSeparatorIndex - 1;
+        const cmdIndex = args.length - afterSeparatorCount;
+        const rawEnvVariables = args.slice(0, -afterSeparatorCount);
+        // Parse environment variables correctly
+        for (const arg of rawEnvVariables) {
+          envVariables.push(binParsers.parseSecretPathEnv(arg));
+        }
+        cmd = args[cmdIndex];
+        argv = args.slice(cmdIndex + 1);
+      }
+      if (envVariables.length === 0) {
+        this.addHelpText('before', 'You must provide at least 1 secret path');
+        this.outputHelp();
+        throw new commander.InvalidArgumentError(
+          'You must provide at least 1 secret path',
         );
+      }
```

Comment:
These are the changes made my this PR. Are these changes okay, or do they need some touch-ups?

Reply:
It looks fine. If there's anything wrong with it I wont spot it from the code alone. We're entering weird territory when it comes to processing the args. So you need to do a bunch of manual testing and check all the edge cases you can think of to make sure it works as expected.

Reply:
All the manual testing I did results in all the tests to pass and also the manual sanity checks I ran to pass. See https://github.com/MatrixAI/Polykey-CLI/pull/336#issuecomment-2489954819.

I also used print statements to print the values of `envVariables`, `cmd`, and `argv`. Nothing stood out from my testing, so I think we can go ahead with the merge.